pkg/chunk/aws/dynamodb_storage_client.go

```diff
@@ -594,6 +590,27 @@ func (a dynamoDBStorageClient) PutChunks(ctx context.Context, chunks []chunk.Chu
 	return a.BatchWrite(ctx, dynamoDBWrites)
 }
 
+func (a dynamoDBStorageClient) DeleteChunk(ctx context.Context, chunkID string) error {
+	userID, err := user.ExtractOrgID(ctx)
+	if err != nil {
+		return err
+	}
```

Comment:
This adds hidden dependency on having userID in the context, which was not there before. If DeleteChunk needs userID, it should be passed explicitly.

Furthermore, it's not really needed -- ParseExternalKey only uses it for verification.

Same comment applies to other `DeleteChunk` implementations in this PR.

Reply:
Changed it. Thanks for the review!